specs/electra/beacon-chain.md

```diff
@@ -1292,43 +1285,62 @@ def process_deposit_receipt(state: BeaconState, deposit_receipt: DepositReceipt)
     )
 ```
 
-##### Consolidations
+##### Execution layer consolidation requests
 
-###### New `process_consolidation`
+###### New `process_execution_layer_consolidation_request`
 
 ```python
-def process_consolidation(state: BeaconState, signed_consolidation: SignedConsolidation) -> None:
-    # If the pending consolidations queue is full, no consolidations are allowed in the block
-    assert len(state.pending_consolidations) < PENDING_CONSOLIDATIONS_LIMIT
-    # If there is too little available consolidation churn limit, no consolidations are allowed in the block
-    assert get_consolidation_churn_limit(state) > MIN_ACTIVATION_BALANCE
-    consolidation = signed_consolidation.message
+def process_execution_layer_consolidation_request(
+    state: BeaconState,
+    execution_layer_consolidation_request: ExecutionLayerConsolidationRequest
+) -> None:
+    # If the pending consolidations queue is full, consolidation requests are ignored
+    if len(state.pending_consolidations) == PENDING_CONSOLIDATIONS_LIMIT:
+        return
+    # If there is too little available consolidation churn limit, consolidation requests are ignored
+    if get_consolidation_churn_limit(state) <= MIN_ACTIVATION_BALANCE:
+        return
+
+    validator_pubkeys = [v.pubkey for v in state.validators]
+    # Verify pubkeys exists
+    request_source_pubkey = execution_layer_consolidation_request.source_pubkey
+    request_target_pubkey = execution_layer_consolidation_request.target_pubkey
+    if request_source_pubkey not in validator_pubkeys:
+        return
+    if request_target_pubkey not in validator_pubkeys:
+        return
+    source_index = ValidatorIndex(validator_pubkeys.index(request_source_pubkey))
+    target_index = ValidatorIndex(validator_pubkeys.index(request_target_pubkey))
+    source_validator = state.validators[source_index]
+    target_validator = state.validators[target_index]
+
     # Verify that source != target, so a consolidation cannot be used as an exit.
-    assert consolidation.source_index != consolidation.target_index
+    if source_index == target_index:
+        return
+
+    # Verify source withdrawal credentials
+    has_correct_credential = has_execution_withdrawal_credential(source_validator)
+    is_correct_source_address = (
+        source_validator.withdrawal_credentials[12:] == execution_layer_consolidation_request.source_address
+    )
+    if not (has_correct_credential and is_correct_source_address):
+        return
+
+    # Verify that target has execution withdrawal credentials
+    if not has_execution_withdrawal_credential(target_validator):
```

Comment:
should add the rewritten version of `assert source_validator.withdrawal_credentials[12:] == target_validator.withdrawal_credentials[12:]`?